pkg/asyncexec/cmd.go

```diff
@@ -0,0 +1,105 @@
+package asyncexec
+
+import (
+	"bufio"
+	"os/exec"
+)
+
+// New instanciate a new Cmd object.
+func New(cmd *exec.Cmd, buff int) *Cmd {
+	return &Cmd{
+		cmd:      cmd,
+		stopChan: make(chan struct{}),
+		StdoutCh: make(chan []byte, buff),
+		StderrCh: make(chan []byte, buff),
+	}
+}
+
+// Cmd is a wrapper arround exec.Cmd. Mainly used to execute
+// command asynchronously with/or without output stream.
+type Cmd struct {
+	cmd *exec.Cmd
+
+	stopChan chan struct{}
+	StdoutCh chan []byte
+	StderrCh chan []byte
+}
+
+// Run runs the command. if streamOutput is true, it will spin
+// two goroutine responsible of streaming the stdout and stderr
+func (c *Cmd) Run(streamOutput bool) error {
```

Comment:
Why not stream output all the time and consumers can just decide whether to read it or not?

Reply:
>consumers can just decide whether to read it or not?

When i first wrote this package i used it to execute multiple command concurrently and i didn't want to leave dozens of channels buffered and open. But i don't think that we're doing that anymore. We can definitely leave it to consumer to decide to read or not from them.